drivers/adc/ad469x/ad469x.c

```diff
@@ -0,0 +1,448 @@
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "stdio.h"
+#include "stdlib.h"
+#include "stdbool.h"
+#include "ad469x.h"
+#include "spi_engine.h"
+#include "delay.h"
+#include "error.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define AD469x_TEST_DATA 0xEA
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+
+/**
+ * Read from device.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad469x_spi_reg_read(struct ad469x_dev *dev,
+			    uint16_t reg_addr,
+			    uint8_t *reg_data)
+{
+	int32_t ret;
+	uint8_t buf[3];
+
+	ret = spi_engine_set_transfer_width(dev->spi_desc, dev->reg_data_width);
+	if (ret != SUCCESS)
+		return ret;
+
+	spi_engine_set_speed(dev->spi_desc, dev->reg_access_speed);
+
+	buf[0] = (1 << 7) | ((reg_addr >> 8) & 0x7F);
+	buf[1] = 0xFF & reg_addr;
+	buf[2] = 0xFF;
+
+	ret = spi_write_and_read(dev->spi_desc, buf, 3);
+	*reg_data = buf[1];
+
+	ret = spi_engine_set_transfer_width(dev->spi_desc,
+					    dev->capture_data_width);
+	if (ret != SUCCESS)
+		return ret;
+
+	spi_engine_set_speed(dev->spi_desc, dev->spi_desc->max_speed_hz);
+
+	return ret;
+}
+
+/**
+ * Write to device.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @@eturn SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad469x_spi_reg_write(struct ad469x_dev *dev,
+			     uint16_t reg_addr,
+			     uint8_t reg_data)
+{
+	int32_t ret;
+	uint8_t buf[3];
+
+	ret = spi_engine_set_transfer_width(dev->spi_desc, dev->reg_data_width);
+	if (ret != SUCCESS)
+		return ret;
+
+	spi_engine_set_speed(dev->spi_desc, dev->reg_access_speed);
+
+	buf[0] = ( (~(1 << 7)) & ((reg_addr >> 8) & 0x7F) );
```

Comment:
```suggestion
	buf[0] = ((reg_addr >> 8) & 0x7F);
```
 (~(1 << 7)) is the same as 0x7F, you don't need to apply the mask twice